src/cast_core/src/AssertiveActor.cpp

```diff
@@ -0,0 +1,216 @@
+        return equalBSONDocs(config, expectedDoc, actualDoc);
+    }
+    return actualVal.get_value() == expectedVal.get_value();
+}
+
+bool equalBSONDocs(AssertiveActor::PhaseConfig* config, const bsoncxx::document::view& expected, const bsoncxx::document::view& actual) {
+    for (auto actualVal : actual) {
+        auto key = actualVal.key();
+        if (key == "_id") {
+            continue;
+        }
+        if (key == "num") {
+            continue;
+        }
+        auto expectedVal = expected[key];
+        if (!equal<bsoncxx::document::element>(config, expectedVal, actualVal)) {
+            BOOST_LOG_TRIVIAL(info) << prefix(config) << "Documents differ in field: " << key;
+            return false;
+        }
+    }
+    for (auto expectedVal: expected) {
+        auto key = expectedVal.key();
+        if (key == "_id") {
+            continue;
+        }
+        if (key == "num") {
+            continue;
+        }
+        if (!actual[key]) {
+            BOOST_LOG_TRIVIAL(info) << prefix(config) << "Documents differ in field: " << key;
+            return false;
+        }
+    }
+    return true;
+}
+
+bool equalBSONArrays(AssertiveActor::PhaseConfig* config, const bsoncxx::array::view& expected, const bsoncxx::array::view& actual) {
+    auto expectedArrLen = std::distance(expected.begin(), expected.end());
+    auto actualArrLen = std::distance(actual.begin(), actual.end());
+    if (expectedArrLen != actualArrLen) {
+        BOOST_LOG_TRIVIAL(info) << prefix(config) << "Arrays differ in length. Expected " << expectedArrLen << " but found " << actualArrLen;
+        return false;
+    }
+
+    auto expectedIt = expected.begin();
+    int count = 0;
+    for (auto actualVal : actual) {
+        auto expectedVal = *expectedIt;
+        if (!equal<bsoncxx::array::element>(config, expectedVal, actualVal)) {
+            BOOST_LOG_TRIVIAL(info) << prefix(config) << "Arrays differ in idx " << count;
+            return false;
+        }
+        ++expectedIt;
+        ++count;
+    }
+
+    return true;
+}
+
+void AssertiveActor::run() {
+    for (auto&& config : _loop) {
+        for (const auto&& _ : config) {
+            auto expected = config->expectedExpr();
+            auto actual = config->actualExpr();
+
+            auto assertOp = _assert.start();
+            try {
+                auto expectedRes = runCommandAndGetResult(config, expected);
+                auto actualRes = runCommandAndGetResult(config, actual);
+                if (equalBSONArrays(config, getBatchFromCommandResult(expectedRes.view()), getBatchFromCommandResult(actualRes.view()))) {
+                    BOOST_LOG_TRIVIAL(info) << prefix(config) << "passed: results are equal.";
+                    assertOp.success();
+                } else {
+                    assertOp.failure();
+                    BOOST_THROW_EXCEPTION(AssertFailed("failed: results were unequal."));
+                }
+            } catch (AssertFailed& e) {
```

Comment:
Would it make sense to have a setting to just record the failure with `assertOp.failure()` but not throw? There are a few such configs for CrudActor: https://github.com/mongodb/genny/blob/30fff93566de289bc83e828cdc3f37c2ceaaefb6/src/cast_core/src/CrudActor.cpp#L46